packages/hosted-form-v2/src/hosted-form-service.ts

```diff
@@ -0,0 +1,22 @@
+import HostedForm from './hosted-form';
+import HostedFormFactory from './hosted-form-factory';
+import HostedFormOptions from './hosted-form-options';
+
+export default class HostedFormService {
+    protected _hostedForm?: HostedForm;
+    constructor(protected _host: string, protected _hostedFormFactory: HostedFormFactory) {}
+
+    initialize(options: HostedFormOptions): Promise<void> {
+        const form = this._hostedFormFactory.create(this._host, options);
+
+        return form.attach().then(() => {
+            this._hostedForm = form;
+        });
+    }
+
+    deinitialize() {
+        if (this._hostedForm) {
+            this._hostedForm.detach();
```

Comment:
Should we unset `_hostedForm` after the detach call? i.e.:

```ts
this._hostedForm = undefined;
```

Otherwise, we will be holding onto a reference of an object that is no longer needed.